core/src/desul/atomics/Compare_Exchange_MSVC.hpp

```diff
@@ -18,6 +18,16 @@ SPDX-License-Identifier: (BSD-3-Clause)
 
 namespace desul {
 
+// Forward declare these functions. They use compare_exchange themselves
+// so the actual header file with them comes after this file is included.
+namespace Impl {
+template <typename MemoryScope>
+inline bool lock_address(void* ptr, MemoryScope ms);
+
+template <typename MemoryScope>
+void unlock_address(void* ptr, MemoryScope ms);
+}
```

Comment:
I am having trouble follow that code spaghetti.
Remind me why we can't just include `Lock_Array.hpp`

Reply:
as the comment above said: Lock_Array.hpp relies on compare_exchange itself, so you can't include it before defining compare_exchange.